src/spriteloader/grf.cpp

```diff
@@ -66,7 +66,8 @@ bool DecodeSingleSprite(SpriteLoader::Sprite *sprite, SpriteFile &file, size_t f
 	 *
 	 * So, any sprite data more than 64 MiB is way larger that we would even expect; prevent allocating more memory!
 	 */
-	if (num < 0 || num > 64 * 1024 * 1024) return WarnCorruptSprite(file, file_pos, __LINE__);
+	constexpr int64 MAX_SPRITE_SIZE = 64 * 1024 * 1024;
+	if (num < 0 || num > MAX_SPRITE_SIZE) return WarnCorruptSprite(file, file_pos, __LINE__);
```

Comment:
Funnily enough the new code requires more instructions commands than the old code when optimisations are disabled. This is mostly to assign that value to some local variable. Making the `constexpr` static removes these extra assembly instructions.
https://godbolt.org/z/sefEv6h79

If you try other compilers, you'll see that Clang, GCC and MSVC all put the precalculated constant in the assembly when optimisations are essentially disabled (-O0). With just a tiny bit of optimsation, the created code is exactly the same.